OOP/Assignment33/RailBooking.java

```diff
@@ -0,0 +1,284 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RailBooking
+{
+	public RailBooking()
+	{
+		setupTrains(); // will fetch all train details and update the list ticketDataList
+		ticketIdCounter = 108; // setting intial ticket id counter to start from 108
+	}
+
+	private List<Train> trainDataList = new ArrayList<Train>();
+	private List<Ticket> ticketDataList = new ArrayList<Ticket>();
+	private final String TRAIN_DATA_FILE = "D:\\TrainData.txt";
+	private int ticketIdCounter;
+
+	// will read all train data from file
+	private void setupTrains()
+	{
+		BufferedReader br;
+		try
+		{
+			br = new BufferedReader(new FileReader(TRAIN_DATA_FILE));
+			String sCurrentLine;
+			while ((sCurrentLine = br.readLine()) != null)
+			{
+				String[] dataFileData = sCurrentLine.split(",");// Reading up questions line by line
+				if (dataFileData.length == 8)
+				{
+					// new Train(id, name, tType, quota, fare, startPos, endPos, tTime);
+					trainDataList.add(new Train(Integer.parseInt(dataFileData[0].trim()), dataFileData[1].trim(), dataFileData[2].trim(), Integer.parseInt(dataFileData[3].trim()), Integer.parseInt(dataFileData[4].trim()), dataFileData[5].trim(), dataFileData[6].trim(), Integer.parseInt(dataFileData[7].trim())));
+				}
+			}
+
+		}
+		catch (IOException e1)
+		{
+			e1.printStackTrace();
+		}
+	}
+
+	public List<Ticket> getTickets()
+	{
+		return ticketDataList;
+	}
+
+	// will throw available trains between start and destination
+	public List<Train> getTrains(String startPos, String DestPos, String TrainType)
+	{
+		List<Train> trainsAvail = new ArrayList<Train>();
+
+		for (Train train : trainDataList)
+		{
+			if (train.getTrainType().toUpperCase().equals(TrainType.trim().toUpperCase()) && train.getTrainStartPos().trim().toUpperCase().equals(startPos.toUpperCase().trim()) && train.getTrainDestPos().trim().toUpperCase().equals(DestPos.toUpperCase().trim()))
```

Comment:
use equalsignorecase() method, so that no need to use toUpperCase method two times.